platform/admin/core/admin-core-configpolicy/src/main/java/org/codice/ddf/admin/config/policy/AdminConfigPolicy.java

```diff
@@ -0,0 +1,172 @@
+    public KeyValueCollectionPermission isPermittedMatchAll(
+            CollectionPermission subjectAllCollection,
+            KeyValueCollectionPermission matchAllCollection) {
+        return isPermittedMatchOne(subjectAllCollection, matchAllCollection);
+    }
+
+    @Override
+    public KeyValueCollectionPermission isPermittedMatchOne(
+            CollectionPermission subjectAllCollection,
+            KeyValueCollectionPermission matchOneCollection) {
+
+        if (matchOneCollection.getAction() == null || (!matchOneCollection.getAction()
+                .equals(VIEW_FEATURE_ACTION) && !matchOneCollection.getAction()
+                .equals(VIEW_SERVICE_ACTION))) {
+            return matchOneCollection;
+        }
+
+        List<Permission> newMatchOneCollectionPermissions =
+                new ArrayList<>(matchOneCollection.getPermissionList());
+        for (Permission permission : matchOneCollection.getPermissionList()) {
+
+            if (!(permission instanceof KeyValuePermission)) {
+                continue;
+            }
+
+            String matchPermissionName = ((KeyValuePermission) permission).getKey();
+
+            Map<String, List<KeyValueCollectionPermission>> policyPermissions;
+
+            if (matchPermissionName.equals(FEATURE_NAME)) {
+                policyPermissions = featurePolicyPermissions;
+            } else if (matchPermissionName.equals(SERVICE_PID)) {
+                policyPermissions = servicePolicyPermissions;
+            } else {
+                continue;
+            }
+
+            boolean isPermitted = true;
+
+            //Typically only one feature or service is desired to be permitted at a time but there is support for multiple feature
+            //If there are multiple features in the permission and one is not authorized, the user is not authorized to see any of the features in the group
+            for (String matchPermissionValue : ((KeyValuePermission) permission).getValues()) {
+                List<KeyValueCollectionPermission> matchOneAttributes = policyPermissions.get(
+                        matchPermissionValue);
+
+                //If null, there is no configuration with this attribute in the policy, the feature or service is white listed
+                if (matchOneAttributes != null) {
+                    boolean matchedOneAttribute = false;
+
+                    for (KeyValueCollectionPermission attributePermissions : matchOneAttributes) {
+                        if (subjectAllCollection.implies(attributePermissions)) {
+                            matchedOneAttribute = true;
+                            break;
+                        }
+                    }
+
+                    if (!matchedOneAttribute) {
+                        isPermitted = false;
+                        break;
+                    }
+                }
+            }
+
+            if (isPermitted) {
+                newMatchOneCollectionPermissions.remove(permission);
```

Comment:
It would be much safer to remove the `isPermitted` boolean and move this removal up to where you're setting the boolean. Then you don't need to worry about the method exiting early and permitting a permission just because it never explicitly denied it.